### PR TITLE
AWS Lambda integration test fix

### PR DESCRIPTION
**Pull request by marcomagdy**
Description:
Fix Lambda integration test.

Risk Level: Low
Testing: unit/integration tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by marcomagdy** [comment]
Looks like the Lambda integration tests passed.
The coverage tests are failing for other reasons. I'll merge from master and retest.

**Comment by mattklein123** [comment]
The coverage failure is a known issue. This change LGTM in general. LMK when ready for review.

File: test/extensions/filters/http/aws_lambda/aws_lambda_filter_integration_test.cc [hunk 1]
```diff
@@ -18,7 +19,13 @@ class AwsLambdaFilterIntegrationTest : public testing::TestWithParam<Network::Ad
   AwsLambdaFilterIntegrationTest()
       : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, GetParam()) {}
 
-  void SetUp() override { setUpstreamProtocol(FakeHttpConnection::Type::HTTP1); }
+  void SetUp() override {
+    // set these environment variables to quickly sign credentials instead of attempting to query
```
Comment (mattklein123) [inline test/extensions/filters/http/aws_lambda/aws_lambda_filter_integration_test.cc]: nit: start sentence with capital, end with period. Same below.

File: source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc [hunk 2]
```diff
@@ -225,9 +225,7 @@ Http::FilterDataStatus Filter::decodeData(Buffer::Instance& data, bool end_strea
   }
 
   auto& hashing_util = Envoy::Common::Crypto::UtilitySingleton::get();
-  if (!decoder_callbacks_->decodingBuffer()) {
-    decoder_callbacks_->addDecodedData(data, false);
-  }
+  decoder_callbacks_->addDecodedData(data, false);
```
Comment (mattklein123) [inline source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc]: Can you also cover this in a unit test and make sure you have the right mock expectations? This should definitely cause a UT failure.

Comment (marcomagdy) [reply source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc]: OK. The integration test fails without this change (as you expected).
I'll add the expectation in the unit tests.

File: test/extensions/filters/http/aws_lambda/aws_lambda_filter_test.cc [hunk 3]
```diff
@@ -167,7 +169,12 @@ TEST_F(AwsLambdaFilterTest, DecodeDataShouldSign) {
   const auto header_result = filter_->decodeHeaders(headers, false /*end_stream*/);
   EXPECT_EQ(Http::FilterHeadersStatus::StopIteration, header_result);
   Buffer::OwnedImpl buffer;
-  ON_CALL(decoder_callbacks_, decodingBuffer()).WillByDefault(Return(&buffer));
+
+  InSequence seq;
+  EXPECT_CALL(decoder_callbacks_, addDecodedData(_, false)).Times(1);
```
Comment (mattklein123) [inline test/extensions/filters/http/aws_lambda/aws_lambda_filter_test.cc]: nit: for future reference `Times(1)` is redundant and not needed. Feel free to remove in a subsequent change.

**Pull request merged**